featuretools/entityset/deserialize.py

```diff
@@ -132,29 +139,54 @@ def read_entity_data(description, path):
 
 
 def read_data_description(path):
-    '''Read data description from disk.
+    '''Read data description from disk, S3 path, or URL.
 
         Args:
-            path (str): Location on disk to read `data_description.json`.
+            path (str): Location on disk, S3 path, or URL to read `data_description.json`.
 
         Returns:
             description (dict) : Description of :class:`.EntitySet`.
     '''
+
     path = os.path.abspath(path)
     assert os.path.exists(path), '"{}" does not exist'.format(path)
     file = os.path.join(path, 'data_description.json')
-    with open(file, 'r') as file:
+    with open(file, 'r', encoding='utf-8') as file:
         description = json.load(file)
     description['path'] = path
     return description
 
 
-def read_entityset(path, **kwargs):
-    '''Read entityset from disk.
+def read_entityset(path, profile_name=None, **kwargs):
+    '''Read entityset from disk, S3 path, or URL.
 
         Args:
-            path (str): Directory on disk to read `data_description.json`.
+            path (str): Directory on disk, S3 path, or URL to read `data_description.json`.
+            profile_name (str): The AWS profile specified to write to S3. Default: None
```

Comment:
boto will detect aws credentials on the machine and use those if no profile name is provided.  I think we should adopt something similar:
* If `profile_name` is a string, use that profile name
* If `profile_name` is `None`, see if boto is able to find credentials and if so use those.  Otherwise use an anonymous profile.
* If `profile_name` is False, use an anonymous profile